kornia/geometry/conversions.py

```diff
@@ -82,7 +82,8 @@ def convert_points_from_homogeneous(points: torch.Tensor) -> torch.Tensor:
 
     # we check for points at infinity
     z_vec: torch.Tensor = points[..., -1:]
-    scale: torch.Tensor = torch.tensor(1.) / torch.clamp(z_vec, EPS)
+    z_vec = torch.where(torch.abs(z_vec) > EPS, z_vec, torch.sign(z_vec) * EPS)
```

Comment:
Is it `* EPS` or could be `* 1`?

Reply:
I think it should be EPS. 
Say we have a homogeneous point, (1, 1, 1), and it corresponding 2D point is (1, 1).
If we devide the point by a huge number(for example, 1e6), the homogeneous point may become (1e-6, 1e-6, 1e-6), and this new homogeneous point should also represent 2D point (1, 1). If we directly set the last element to 1, the corresponding 2D point would become (1e-6, 1e-6), which is wrong.

Reply:
Yes for this case could be ok. I've opened a cross metnion at https://github.com/opencv/opencv/pull/14411#issuecomment-496677389 cause we had open the EPS issue also on Opencv.

Reply:
In theory, Z=0 case is point at infinity. `(0, 0, 0)` doesn't represent any point.
In computer vision tasks these points are not out cases.

Homogeneous points with near-zero Z are bad too (usually they are result of non converge algorithm and/or other floating-point calculation errors).
In "ideal world" it is better to filter out such outlier points/cases with proper processing.
OpenCV's trick just tries to avoid INF/NANs in results.

Note about the current proposal:
Should `(1e-9, 1e-9, 1e-9)` represent 2D `(1000, 1000)`? (EPS=1e-6, but 2D result depends on EPS value which is not good in general).

Reply:
@alalek Yes, also here especially (pytorch)  we need to maintain some form of differentiation and so we need to avoid to have NAN for derivatives for autodiff.

Reply:
I see, the 2D point position would change depends on the EPS value.
I agree with that the position should not depends on EPS.
Let's stick with opencv impelemetation and set near-zero values to 1?

Reply:
Yes. Could you add a note in the docs?

Reply:
Added some comments in the code file

Reply:
great! and thanks @alalek for the comment :D